deploy/tests/util/test_git.py

```diff
@@ -0,0 +1,159 @@
+import logging
+from os import fspath
+import pytest
+import re
+from subprocess import run, PIPE
+import sure
+
+from util.streamlogger import InfoStreamLogger
+
+
+rootLogger=logging.getLogger()
+
+class TestGitAssumptions:
+    """Test git features that we depend on"""
+
+    def test_git_fetch_pack_for_unadvertised_sha(self):
```

Comment:
@abejgonzalez can you help me understand why we need to create a second-level of clone in the CI scripts. See https://github.com/firesim/firesim/blob/0552050b95983be81bb71d03408241d0e3bfb783/.github/scripts/initialize-repo.py#L14-L15

This test is checking that the behavior of github (or whatever server the project is using to host the repo) but it assumes that we're in a clone that has origin set to the server we care about.

The test currently passes but it isn't testing anything because the clone it is running in was cloned from another directory on the CI host and not the actual github repository.